src/elf/sym.rs

```diff
@@ -163,93 +163,92 @@ macro_rules! elf_sym_std_impl {
-                    Sym {
-                        st_name:     sym.st_name as u32,
-                        st_info:     sym.st_info,
-                        st_other:    sym.st_other,
-                        st_shndx:    sym.st_shndx as u16,
-                        st_value:    sym.st_value as $size,
-                        st_size:     sym.st_size as $size,
-                    }
+        impl From<ElfSym> for Sym {
+            #[inline]
+            fn from(sym: ElfSym) -> Self {
+                Sym {
+                    st_name:     sym.st_name as u32,
+                    st_info:     sym.st_info,
+                    st_other:    sym.st_other,
+                    st_shndx:    sym.st_shndx as u16,
+                    st_value:    sym.st_value as $size,
+                    st_size:     sym.st_size as $size,
                 }
             }
+        }
 
-            impl fmt::Debug for Sym {
-                fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-                    let bind = st_bind(self.st_info);
-                    let typ = st_type(self.st_info);
-                    let vis = st_visibility(self.st_other);
-                    f.debug_struct("Sym")
-                        .field("st_name", &self.st_name)
-                        .field("st_value", &format_args!("{:x}", self.st_value))
-                        .field("st_size", &self.st_size)
-                        .field("st_info", &format_args!("{:x} {} {}", self.st_info, bind_to_str(bind), type_to_str(typ)))
-                        .field("st_other", &format_args!("{} {}", self.st_other, visibility_to_str(vis)))
-                        .field("st_shndx", &self.st_shndx)
-                        .finish()
-                }
+        impl fmt::Debug for Sym {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                let bind = st_bind(self.st_info);
+                let typ = st_type(self.st_info);
+                let vis = st_visibility(self.st_other);
+                f.debug_struct("Sym")
+                    .field("st_name", &self.st_name)
+                    .field("st_value", &format_args!("{:x}", self.st_value))
+                    .field("st_size", &self.st_size)
+                    .field("st_info", &format_args!("{:x} {} {}", self.st_info, bind_to_str(bind), type_to_str(typ)))
+                    .field("st_other", &format_args!("{} {}", self.st_other, visibility_to_str(vis)))
+                    .field("st_shndx", &self.st_shndx)
+                    .finish()
             }
+        }
 
-            #[inline]
-            pub unsafe fn from_raw<'a>(symp: *const Sym, count: usize) -> &'a [Sym] {
-                slice::from_raw_parts(symp, count)
-            }
+        #[inline]
+        pub unsafe fn from_raw<'a>(symp: *const Sym, count: usize) -> &'a [Sym] {
+            slice::from_raw_parts(symp, count)
+        }
+
+        if_alloc! {
+            if_std! {
```

Comment:
I think you can delete the `if_alloc! {}` if everything inside it is gated by `if_std!`.

Reply:
Yeah, that's right